src/app/services/auth.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, inject, signal } from '@angular/core';
+import {
+  Auth,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+  updateProfile,
+  user,
+} from '@angular/fire/auth';
+import { Observable, from } from 'rxjs';
+import { IUser } from '../models/user';
+import { CrudService } from './crud.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  firebaseAuth = inject(Auth);
+  crudService = inject(CrudService);
+  user$ = user(this.firebaseAuth); // persisting user when logged in
+  currentUserSignal = signal<IUser | null | undefined>(undefined); // undefined because it takes time to get the user
+
+  register(
+    email: string,
+    name: string,
+    surname: string,
+    password: string
+  ): Observable<void> {
+    let newUser: IUser = {
+      email: email,
+      name: name,
+      surname: surname,
+      password: password,
+      userId: '',
+    };
+
+    const promise = createUserWithEmailAndPassword(
+      this.firebaseAuth,
+      email,
+      password
+    )
+      .then((response) => {
+        updateProfile(response.user, { displayName: name });
+        newUser.userId = response.user.uid;
+        this.crudService.addUser(newUser);
+      })
+      .catch((error) =>
+        console.error('Error with registering the user.', error)
+      );
+
+    return from(promise);
+  }
+
+  login(email: string, password: string): Observable<void> {
+    const promise = signInWithEmailAndPassword(
```

Comment:
Thank you but I think I prefer to have it as `return from(promise)` so that when i come back to it i immediately know whats happening especially while i am still going to refractor it to fit into the other parts of the system